### PR TITLE
support filter on selector dropdown

### LINKED ISSUE #428: Possibility to add a filter to a selector even if dropdown is True
**Description**

When filter is True and dropdown False, a filter section appears to filter the selection.

It would be great if this also works when dropdown is True. Right now, no filter section is visible.

```python
from taipy import Gui 

lov = ['a','b','c']
selected = ''

Gui('<|{selected}|selector|dropdown|filter|lov={lov}|>').run(port=5001)
```

image

**Acceptance Criteria**
- [ ] Ensure new code is unit tested, and check code coverage is at least 90%
- [ ] Propagate any change on the demos and run all of them to ensure there is no breaking change
- [ ] Ensure any change is well documented

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
update date-fns to 3.x
resolves #428

```
from taipy.gui import Gui, State

lov = ['node_modules/uuid/dist/esm-browser','src/context','webpack 5.91.0 compiled successfully in 2742 ms']
selected = 'src/context'

def on_change(state: State, var: str, val):
  print(f"on_change(state, var: {var}, val: {val})")

Gui("""<|{selected}|selector|dropdown|filter|lov={lov}|label=dropdown+filter|>
    <|{selected}|selector|dropdown|mode=check|lov={lov}|label=dropdown+check|>
    <|{selected}|selector|dropdown||filter|lov={lov}|label=dropdown+filter|>
    <|{selected}|selector|dropdown|multiple|filter|lov={lov}|label=dropdown+multiple+filter|>
    <|{selected}|selector|dropdown|multiple|don't filter|lov={lov}|label=dropdown+multiple|>
    """).run()

```

### CODE REVIEW

**Review by dinhlongviolin1 — DISMISSED** [review]
LGTM! Really straight forward

**Review by namnguyen20999 — DISMISSED** [review]
LGTM

File: taipy/gui/_renderers/builder.py [hunk 1]
```diff
@@ -827,7 +827,10 @@ def set_value_and_default(
                             value = float(value)
                     if isinstance(value, (int, float)):
                         return self.__set_react_attribute(_to_camel_case(var_name), value)
-                self.set_attribute(_to_camel_case(var_name), value)
+                if isinstance(value, (datetime, date, time)):
```
Comment (FabienLelaquais) [inline taipy/gui/_renderers/builder.py:830]: I would have wriitten this:
```suggestion
                if isinstance(value, (datetime, date, time)):
                    value = _date_to_string(value)
                self.set_attribute(_to_camel_case(var_name), value)
```
That to me is more readable.

**Review by FabienLelaquais — DISMISSED** [review]
👌

One thing that needs to be done is reorder the properties in viselements.json.
Order should be
label, mode, dropdown, multiple, filter, width and height

**Review by FabienLelaquais — APPROVED** [review]
👌

**Review by namnguyen20999 — APPROVED** [review]

**Pull request merged**